ai-in-directus-flows/index.md

```diff
@@ -0,0 +1,154 @@
+---
+title: 'No code AI with Directus Flows'
+description: ‘Learn how AIVenture used Directus Flows to create a low-code, AI based, text-adventure game engine in the style of the classic Zork games.’
+author:
+  name: 'Craig Harman'
+  avatar\_file\_name: 'craig.jpg'
+---
+
+## You are standing in an open field…
+
+[AIVenture][1] hark's back to the days of text-based adventure games such at [Zork][2] and [Hitchhiker’s Guide to the Galaxy][3] but with a modern AI twist. The game uses Directus Flows and user registration along with the [directus-extension-ai-pack][4] extension to create a low-code game engine offering players a unique story-telling experience.
+
+The underlying application is effectively a Chat-GPT powered game engine. It uses Directus to control game logic, provide player registration, and track game state persistence. The front-end is a [Nuxt][5] application which receives player commands (in the form of text prompts) and sends them to a Directus Flow endpoint. Responses are generated by Chat-GPT using the [directus-extension-ai-pack][6] extension.
+
+## On boarding players
+So that AIVenture can store game progress across devices individual users need to be identified. 
+
+As Directus already has user management and authentication baked in AIVenture is able to harness the API to create users. When users sign up they are assigned a custom “player” role that gives them permissions to reach a single Directus Flow game endpoint.
+
+To avoid the game sign-up process being spammed with fake email addresses we require that a player’s email address is validated. Although there isn’t specifically an email validation workflow available in Directus, there is a [user invite procedure][7]. Behind the scenes when a player signs up to AIVenture we are actually generating a user **invite** in Directus. This does not give access to the game until it has been accepted via a link in the player’s email - which is effectively the same as an email validation process.
```

Comment:
```suggestion
To avoid the game sign-up process being spammed with fake email addresses, we require that a player’s email address is validated. Although there isn’t specifically an email validation workflow available in Directus, there is a [user invite procedure][7]. Behind the scenes when a player signs up to AIVenture we are actually generating a user **invite** in Directus. This does not give access to the game until it has been accepted via a link in the player’s email - which is effectively the same as an email validation process.
```